dbms/tests/queries/0_stateless/01019_array_fill.sql

```diff
@@ -0,0 +1,11 @@
+SELECT arrayFill(x -> 0, [1, 2, 3, 11, 12, 13, 4, 5, 6, 14, 15, 16]);
```

Comment:
I'm not sure I got that right, but it seems to me that the function name is a little misleading because it doesn't fill anything in, it's probably more like `arrayMapWithIndex` ?

For example:
``` sql
SELECT arrayMapWithIndex((element, index) -> index, [1, 2, 3, 11, 12, 13, 4, 5, 6, 14, 15, 16]) = [1, 2, 3, 11, 12, 13, 4, 5, 6, 14, 15, 16];

SELECT arrayMapWithIndex((element, index) -> index == 0 ? 0: index - 1, [1, 2, 3, 11, 12, 13, 4, 5, 6, 14, 15, 16]) = [1, 1, 2, 3, 11, 12, 13, 4, 5, 6, 14, 15];
```

Reply:
This test case is to cover a corner case (when the lambda yields a constant value)